lib/build.js

```diff
@@ -443,13 +443,14 @@ function createWebpackMiddleware () {
   const clientCompiler = webpack(clientConfig)
   this.clientCompiler = clientCompiler
   // Add the middleware to the instance context
-  this.webpackDevMiddleware = pify(require('webpack-dev-middleware')(clientCompiler, {
+  this.webpackDevMiddleware = pify(require('webpack-dev-middleware')(clientCompiler, Object.assign(clientConfig.devServer||{},{
```

Comment:
Yes, devServer is a standard option, and is designed to be passed into dev-middleware.  Dev-middleware is the package that provides the devServer functionality.  As for not-mutating options, I'm onboard - I just would love it if some way to pass in devServer details to the underlying thing.